packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -716,21 +716,27 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
     // Special case: There's nothing to work on.
     if (existingCallbackNode !== null) {
       cancelCallback(existingCallbackNode);
-      root.callbackNode = null;
-      root.callbackPriority = NoLanePriority;
     }
+    root.callbackNode = null;
+    root.callbackPriority = NoLanePriority;
     return;
   }
 
   // Check if there's an existing task. We may be able to reuse it.
-  if (existingCallbackNode !== null) {
-    const existingCallbackPriority = root.callbackPriority;
-    if (existingCallbackPriority === newCallbackPriority) {
-      // The priority hasn't changed. We can reuse the existing task. Exit.
-      return;
+  const existingCallbackPriority = root.callbackPriority;
+  if (existingCallbackPriority === newCallbackPriority) {
+    if (__DEV__) {
+      invariant(
+        existingCallbackNode,
```

Comment:
This seems wrong? `existingCallbackNode` will be null if the existing task is a microtask.

Reply:
Side note: We shouldn't have dev-only invariants. Use a warning so the semantics don't diverge. Though because of my previous point, the warning shouldn't exist, either.

Reply:
Yeah, was just fixing this.